src/core/ext/filters/client_channel/resolving_lb_policy.cc

```diff
@@ -306,61 +309,45 @@ void ResolvingLoadBalancingPolicy::OnResolverResultChangedLocked(
   //
   // We track a list of strings to eventually be concatenated and traced.
   TraceStringVector trace_strings;
-  const bool resolution_contains_addresses = result.addresses.size() > 0;
-  // Process the resolver result.
-  ChannelConfigHelper::ApplyServiceConfigResult service_config_result;
+  MaybeAddTraceMessagesForAddressChangesLocked(!result.addresses.empty(),
+                                               &trace_strings);
+  // Hold a ref to the error so that the entry in trace_strings lives
```

Comment:
I don't understand this comment but since Resolver::Result::~Result() unrefs `service_config_error`, I'm not worried about the error being unreffed atleast.

Reply:
The result of `grpc_error_string()`, which we're storing in `trace_strings`, is a pointer to a string that is owned by the error itself.  However, we transfer ownership of the `Resolver::Result` object containing the error to the LB policy on line 336, and the LB policy will destroy it before that call returns.  So we need to hold a ref to the error until line 349, where we're actually using `trace_strings`.

I've tried to clarify the comment here.